### ISSUE TITLE
Layout completely broken

### ISSUE THREAD

**Issue by chrjsorg** [roughike/BottomBar#126]
Do you have any idea how this layout is possible?
http://i.imgur.com/Rn2kCOC.png

This is the layout xml:
https://gist.github.com/chrjsorg/83e5f11af0676afc7fef67e508a656fe

I tried attach() and attachShy() with the CoordinatorLayout + ScollLayout (null) and I'm calling .noTopOffset()

It happens on my Nexus 5X. On the emulator its looking fine.

**Comment by roughike** [comment]
How are you using it in Java?

**Comment by chrjsorg** [comment]
I tried multiple things (not all at once obviously :))

```
bottomBar = BottomBar.attach(this, savedInstanceState) 
bottomBar = BottomBar.attachShy((CoordinatorLayout) findViewById(R.id.coordinatorLayout_main_appBar), null, savedInstanceState));
bottomBar = BottomBar.attach(findViewById(R.id.coordinatorLayout_main_appBar), savedInstanceState));
bottomBar.get().noTopOffset();
```

**Comment by roughike** [comment]
Are you sure that's all? You're not using ```setDefaultTabPosition(2)``` or ```selectTabAtPosition()``` methods?

**Comment by chrjsorg** [comment]
Its complex...
I'm calling all of the above + setDefaultTabPosition(my_synced_tab_position) onResume. I dont have any savedInstanceState (its always null) while attaching.

I have multiple Activities using its own BottomBar, so I try so synchronize the selected bottom tabs and have to select it manually on onResume.

**Comment by roughike** [comment]
I'm fixing this as we speak. The issue is related to setDefaultTabPosition / selectTabAtPosition methods, which also update the widths of the tabs if the resize animation is enabled.

**Comment by roughike** [comment]
Try with the new _1.2.8._ version.

**Comment by chrjsorg** [comment]
It looks better now. But its still not perfect. Maybe because I can't call attach with the savedInstanceState? :/

http://i.imgur.com/aJwXdtZ.png

http://i.imgur.com/jfScTTI.png

**Comment by roughike** [comment]
Is there a reason why you don't use one Activity with multiple Fragments instead of multiple Activities with multiple BottomBars?

**Comment by chrjsorg** [comment]
Unfortunately I can't change that for the moment. Maybe it will be required anyway since the animations will be broken anyway (probably). 
I know our use case isn't the best..

**Comment by cyberjaime45** [comment]
I had the same the same issue. I though it was due to my device was running Android N. But then I used a marshmallow  device, same thing. The layout gets mess up when I added this to resource file fitsSystemWindows=true. So I removed, and added some extra padding  to custom toolbar. I also added mBottomBar.noTopOffset() before   mBottomBar.setItemsFromMenu(.......

Its kind tacky but it works just fine...
